core/src/main/java/org/infinispan/persistence/factory/DeployedCacheStoreFactory.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.persistence.factory;
+
+import org.infinispan.configuration.cache.StoreConfiguration;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.infinispan.persistence.factory.ConfigurationForClassExtractor.getClassBasedOnConfigurationAnnotation;
+
+/**
+ * Cache Store factory designed for deployed instances.
+ *
+ * @author Sebastian Laskawiec
+ * @since 7.2
+ */
+public class DeployedCacheStoreFactory implements CacheStoreFactory {
```

Comment:
I don't see why this class should live in `core`. If this class is designed for deployed cache store instances, then it should live in `server/integration/endpoint`